palindrome_checker.cpp

```diff
@@ -1,6 +1,43 @@
+/*
+______________ Assumptions _____________
+1. LinkedList contains int or char
```

Comment:
Reversing and comparing the list is a solid approach and has good time/space complexity. Consider if you can solve the same problem with half as much space